src/Net/Http/HttpOptions.php

```diff
@@ -152,6 +152,125 @@ public function setProxy($proxy)
     }
 
     /**
+     * @return string
+     */
+    public function getUserAgent()
+    {
+        return $this->get('user_agent');
+    }
+
+    /**
+     * @param string $userAgent
+     * @return $this
+     */
+    public function setUserAgent($userAgent)
+    {
+        return $this->set('user_agent', "$userAgent");
+    }
+
+    /**
+     * @return boolean
+     */
+    public function getFollowLocation()
+    {
+        return $this->get('follow_location');
+    }
+
+    /**
+     * @param boolean $followLocation
+     * @return $this
+     */
+    public function setFollowLocation($followLocation)
+    {
+        return $this->set('follow_location', (bool)$followLocation);
+    }
+
+    /**
+     * @return boolean
+     */
+    public function getRequestFullUri()
+    {
+        return $this->get('request_fulluri');
+    }
+
+    /**
+     * @param boolean $requestFullUri
+     * @return $this
+     */
+    public function setRequestFullUri($requestFullUri)
+    {
+        return $this->set('request_fulluri', (bool)$requestFullUri);
+    }
+
+    /**
+     * @return int
+     */
+    public function getMaxRedirects()
+    {
+        return $this->get('max_redirects');
+    }
+
+    /**
+     * @param int $maxRedirects
+     * @return $this
+     */
+    public function setMaxRedirects($maxRedirects)
+    {
+        return $this->set('max_redirects', $maxRedirects|0);
+    }
+
+    /**
+     * @return float
+     */
+    public function getProtocolVersion()
+    {
+        return $this->get('protocol_version');
+    }
+
+    /**
+     * @param float $protocolVersion
+     * @return $this
+     */
+    public function setProtocolVersion($protocolVersion)
+    {
+        return $this->set('protocol_version', $protocolVersion);
```

Comment:
This is missing a cast on `$protocolVersion`. A float cast can be done with [coercion](http://codelegance.com/embracing-coercion-in-php/) or just a regular `(float)` cast. I suggest a regular cast is clearer for most readers. Applies here and again on line 253.